### DISCUSSION TITLE
Funcionalitats del control d'assistència

### DISCUSSION THREAD

**Discussion by FherStk** [ElPuig/IMS#2] [2. Equip de treball: assistència i fitxa d'alumne]
Deixo aquesta discussió preparada per parlar de les funcionalitats concretes (que farà i com ho farà) de la part de "control d'assistència i fitxa d'alumne".

**Comment by FherStk** [comment]
A continuació exposo una idea de com imagino que podria funcionar el sistema de control d'assistència, òbviament obert a discussió:

- Plantilla: un professor pot crear una plantilla d'assistència, en aquesta plantilla determina el següent:
  - MP: el Mòdul Professional que s'imparteix (per a la ESO/BTX poden ser assignatures).
  - UF: la UF que s'imparteix (per a la ESO/BTX poden ser temes, o es pot deixar com a opcional).
  - Aula: en quina aula s'imparteix.
  - Marc horari: hora d'inici i hora final.
  - Professorat: quins professors es troben a l'aula (cal tenir en compte que tenim MPs amb més d'un professor a l'aula).
  - Alumnat: aquí es poden afegir o treure els alumnes que es consideri que han d'assistir. Es disposarà del llistat d'alumnat complet, carregat automàticament del SAGA.

L'Oscar va comentar a la reunió que seria interessant buscar la manera de generar unes plantilles de forma automàtica, per a facilitar l'ús de l'eina al professorat poc experimentat.

Quan un professor vol passar llista, haurà d'escollir la plantilla i li apareixerà tota la informació d'aquesta junt amb les opcions de marcar l'estat de cada alumne a l'aula (assistència, falta, retard, incidència, etc.). Potser és interessant assumir que el professor que passa llista es troba present a l'aula, i deixar que pugui marcar si altre professorat hi és present o no. Això obriria la porta a compartir plantilles entre tots els professors que fan docència a una aula (i, en el cas d'un professor de guàrdia, li hauria de deixar passar llista amb aquesta plantilla, marcant com a present al professor de guàrdia i absent als titulars). 

Respecte dels alumnes, Direcció vol que les famílies rebin notificació de les incidències amb alumnes menors d'edat i que els tutors puguin mantenir viva la fitxa de l'estudiant, emplenant informació i pujant documents (com informes, justificants mèdics, etc.). 

Altres temes o idees:
  - Quan un tutor penja un justificant d'absència a la fitxa d'un alumne, que el sistema justifiqui automàticament les faltes per a aquell conjunt de dies. 
  - Cal valorar quina és la millor manera de gestionar l'avís a les famílies quan hi ha una incidència. L'iEduca ho fa quan se signa el llistat d'assistència, però aleshores qualsevol canvi posterior no es notifica. Es podria fer quelcom similar amb l'opció de tornar a signar i notificar només als que no hagin estat notificats). 
  - Cal valorar si s'aprofitaran els models de dades d'Odoo (com per exemple el Customer) o es faran models propis per a tots. Potser és millor treballar amb models propis per tal de garantir la independència de futurs canvis a Odoo i mantenir el control de les dades així com intentar reduir al màxim el volum de la base de dades. No obstant, però, cal tenir en compte que Odoo fara servir el model "res_partner" per a definir la informació dels usuaris d'Odoo, els clients, els proveidors, etc. Potser disposa de funcionalitats ja implementades que ens puguin venir bé de cara la fitxa de l'alumne. Cal investigar-ho.

**Reply by FherStk** [reply]
Bon dia companys,

Aquest curs que entra fusionaré MP03 i MP05, i he distribuït els continguts per NF de tal manera que cada NF treballa continguts (Cs) de UFs diferents i de MPs diferents. L'avaluació la faré per Criteris d'Avaluació (CAs, que pertanyen a UFs), però el control d'assistència el faré per NF.

Tinc definit, per cada NF, quantes hores toquen de cada UF. Per tant, si es posa una falta dins un NF, es podria calcular quin % de faltes pertoca dins de cada UF. 

Per tant, de cara a una futura iteració del control d'assistència, proposo el següent:
- Que el professor, en el moment de crear la seva plantilla per passar llista dins un grup, pugui escollir de quina forma vol controlar l'assistència:
  - Per UFs: seria la forma estàndard dins de CCFF.
  - Per MP/Assignatura: aquesta opció seria l'adient per a l'ESO/BTX.
  - Per NF: aquesta seria la que faria servir en aquest cas, caldria definir per cada NF quantes hores toca de cada UF.

Faltaria tenir en compte que, en aquest cas, estem parlant de dos MPs que s'estan cursant a l'hora. Això també caldria tenir-ho en compte quan es fa una plantilla, perquè després es pugui desglossar per cada MP/UF de forma individual. 

Salut i bon estiu!
Fer

**Comment by ChusVazquez** [comment]
La part de fer el mateix MP varies vegades (si ho entenc bé, seria, per exemple, MP05 a DAM i a DAW alhora) no hauria de donar problema ja que sempre es farà a diferents grups d'alumnes i podem fer servir aixó per agrupar/desglossar a aquest nivell, oi?

**Reply by FherStk** [reply]
Correcte :)
